lib/bolt/task.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'bolt/error'
+
+module Bolt
+  class Task
+    attr_reader :name, :metadata, :files
+
+    def self.task_name?(name)
+      return true if name =~ /^[a-z][a-z0-9_]*$/
+      false
+    end
+
+    def initialize(task_name, files, metadata = nil)
+      unless Bolt::Task.task_name?(task_name)
+        raise Bolt::Error, _("Task names must start with a lowercase letter " \
+                             "and be composed of only lowercase letters, numbers, and underscores")
+      end
+
+      @name = task_name
+      @metadata = metadata if metadata
```

Comment:
should we validate this if we're validating the name?

Reply:
Created https://tickets.puppetlabs.com/browse/BOLT-658 -- I couldn't find a function that exists  to do this already, and we should probably validate the rest of the data we recieve as well.